### ISSUE TITLE
GLupeN64 segfaults in valgrind

### ISSUE THREAD

**Issue by orbea** [loganmc10/GLupeN64#84]
OS: `Slackware64-current`
`libdrm-2016.09.22_9e24d0c_master-x86_64-1_git`
`mesa-2016.10.19_7b7eb71_master-x86_64-1_git`
`valgrind-3.11.0-x86_64-1_SBo`
`RetroArch-2016.10.23_31180df_master-x86_64-1_git`
`GLupeN64-2016.10.19_e72887c_master-x86_64-1_git`

When running GLupeN64 in valgrind it will crash, this does not happen with mupen64plus + GLideN64. It might be a result of libco?
```
==6759== Process terminating with default action of signal 11 (SIGSEGV)
==6759==  Access not within mapped region at address 0x4
==6759==    at 0x4CBD7A: action_ok_file_load (menu_cbs_ok.c:1940)
==6759==    by 0x4C6D63: menu_entry_action (menu_entry.c:518)
==6759==    by 0x4F4F7E: generic_menu_iterate (menu_generic.c:246)
==6759==    by 0x4AE96C: menu_driver_ctl (menu_driver.c:887)
==6759==    by 0x424E35: runloop_check_state (runloop.c:920)
==6759==    by 0x4257CA: runloop_iterate (runloop.c:1186)
==6759==    by 0x4113CD: rarch_main (frontend.c:127)
==6759==    by 0x411434: main (frontend.c:145)
==6759==  If you believe this happened as a result of a stack
==6759==  overflow in your program's main thread (unlikely but
==6759==  possible), you can try to increase the size of the
==6759==  main thread stack using the --main-stacksize= flag.
==6759==  The main thread stack size used in this run was 8388608.
```
Valgrind log: http://pastebin.com/cf6zyBA5

**Comment by loganmc10** [comment]
mupen64plus-libretro uses libco as well. This will be a little tricky to diagnose since that stacktrace is actually pointing to RetroArch (all of those functions reside in RetroArch, not GLupeN64)

**Comment by orbea** [comment]
A similar crash used to occur in mupen64plus-libretro, but I just checked now and it valgrind works (Slowly) with glide64, gln64, rice and angrylion. I can bisect this if you think it will help?

https://github.com/libretro/mupen64plus-libretro/issues/340

**Comment by loganmc10** [comment]
Bisecting would definitely help, but it has probably existed for a while, it'd probably be helpful to find out what commit fixed it in mupen64plus-libretro

**Comment by orbea** [comment]
There are a few problems with that, first it appears it was never fixed in mupen64plus-libretro. Rather it was fixed in the upgrade from valgrind 3.10.1 to 3.11.0. Additionally this now also seems true for GLupeN64 which works with valgrind 3.11.0 and not 3.10.1, but my log clearly shows I was using 3.11.0 so I'm not sure why I can't reproduce this anymore.

I guess I will close this until it can be reproduced. :)
